### PR TITLE
[RTF-1415] workspace strategy for Update operation

### PR DESCRIPTION
**Pull request by maria-kizhaeva**
Complete the following tasks before merging in your PR:
- [x] If this is a user facing change, update [version.txt](https://github.com/relativitydev/relativity.testing.framework.api/blob/master/version.txt) and [CHANGELOG.md](https://github.com/relativitydev/relativity.testing.framework.api/blob/master/CHANGELOG.md)

### CODE REVIEW

File: source/Relativity.Testing.Framework.Api/Strategies/Workspaces/WorkspaceUpdateStrategyV1.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+					DefaultFileRepository = new Securable<Artifact>(entity.DefaultFileRepository),
+					DownloadHandlerUrl = entity.DownloadHandlerUrl,
+					EnableDataGrid = false,
+					entity.Keywords,
+					entity.Notes,
+					ResourcePool = new Securable<Artifact>(entity.ResourcePool),
+					SqlFullTextLanguage = entity.SqlFullTextLanguage,
+					SqlServer = new Securable<Artifact>(entity.SqlServer),
+					Status = new Artifact(675),
+					Template = entity.TemplateWorkspace == null ? null : new Securable<Artifact>(new Artifact(entity.TemplateWorkspace.ArtifactID)),
```
Comment (paulletang) [inline source/Relativity.Testing.Framework.Api/Strategies/Workspaces/WorkspaceUpdateStrategyV1.cs:46]: Is null a valid value here?

File: source/Relativity.Testing.Framework.Api/Strategies/Workspaces/WorkspaceUpdateStrategyV1.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+					DefaultCacheLocation = new Securable<Artifact>(entity.DefaultCacheLocation),
+					DefaultFileRepository = new Securable<Artifact>(entity.DefaultFileRepository),
+					DownloadHandlerUrl = entity.DownloadHandlerUrl,
+					EnableDataGrid = false,
+					entity.Keywords,
+					entity.Notes,
+					ResourcePool = new Securable<Artifact>(entity.ResourcePool),
+					SqlFullTextLanguage = entity.SqlFullTextLanguage,
+					SqlServer = new Securable<Artifact>(entity.SqlServer),
+					Status = new Artifact(675),
```
Comment (paulletang) [inline source/Relativity.Testing.Framework.Api/Strategies/Workspaces/WorkspaceUpdateStrategyV1.cs:45]: What if someone want to use a different status?

File: source/Relativity.Testing.Framework.Api/Services/IWorkspaceService.cs [hunk 3]
```diff
@@ -111,6 +111,20 @@ public interface IWorkspaceService
 		/// </example>
 		Workspace CreateWithDocs(Workspace entity, int numberOfDocuments = 10);
 
+		/// <summary>
+		/// Updates the specified [Workspace](https://relativitydev.github.io/relativity.testing.framework/api/Relativity.Testing.Framework.Models.Workspace.html).
+		/// </summary>
+		/// <param name="entity">The entity to update.</param>
```
Comment (paulletang) [inline source/Relativity.Testing.Framework.Api/Services/IWorkspaceService.cs:117]: -> `The workspace to update`

File: source/Relativity.Testing.Framework.Api/Strategies/Workspaces/WorkspaceUpdateStrategyV1.cs [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+		public void Update(Workspace entity)
+		{
+			if (entity is null)
+			{
+				throw new ArgumentNullException(nameof(entity));
+			}
+
+			_restService.Put($"relativity-environment/V1/workspace/{entity.ArtifactID}", BuildRequest(entity));
+		}
+
```
Comment (paulletang) [inline source/Relativity.Testing.Framework.Api/Strategies/Workspaces/WorkspaceUpdateStrategyV1.cs:27]: Can we add some unit tests for the code in this strategy?

File: source/Relativity.Testing.Framework.Api/Strategies/Workspaces/WorkspaceUpdateStrategyV1.cs [hunk 5]
```diff
@@ -42,8 +42,7 @@ private object BuildRequest(Workspace entity)
 					ResourcePool = new Securable<Artifact>(entity.ResourcePool),
 					SqlFullTextLanguage = entity.SqlFullTextLanguage,
 					SqlServer = new Securable<Artifact>(entity.SqlServer),
-					Status = new Artifact(675),
-					Template = entity.TemplateWorkspace == null ? null : new Securable<Artifact>(new Artifact(entity.TemplateWorkspace.ArtifactID)),
```
Comment (jonatanjakoniuk) [inline source/Relativity.Testing.Framework.Api/Strategies/Workspaces/WorkspaceUpdateStrategyV1.cs]: I would keep `Template  = null` with a comment that this is API requirement + link to doc page.
In can be not obvious for others devs in future why `Template` property is skipped here.

File: source/Relativity.Testing.Framework.Api/Strategies/Workspaces/WorkspaceUpdateStrategyV1.cs (hunk 2 above)
Comment (maria-kizhaeva) [reply source/Relativity.Testing.Framework.Api/Strategies/Workspaces/WorkspaceUpdateStrategyV1.cs]: I agree it shouldn't stay as it is. I just missed this one taking it straight from creating workflow. 
But the thing is that the API accepts Status Artifact ID and we keep it as String only. We'd need to change Status field from String to NamedArtifact(or Enum) or at least add GetEligibleStatuses strategy and lookup strategy ArtifactId by it's name that we store.

**Pull request merged**